base/uk.ac.stfc.isis.ibex.e4.product/src/uk/ac/stfc/isis/ibex/e4/product/ApplicationWorkbenchWindowAdvisor.java

```diff
@@ -70,21 +106,157 @@ public void preWindowOpen() {
     public void postWindowCreate() {
     	super.postWindowCreate();
         final Shell shell = getWindowConfigurer().getWindow().getShell();
-
+        
         WindowLayout windowLayout = getPreviousWindowSettings();
 
         shell.setMinimumSize(MIN_WINDOW_WIDTH, MIN_WINDOW_HEIGHT);
         shell.setLocation(windowLayout.windowX, windowLayout.windowY);
         shell.setSize(windowLayout.windowWidth, windowLayout.windowHeight);
         shell.setMaximized(windowLayout.windowMaximised);
+        
+        try {
+    		if (clearTempFile().length > 0) {
+    			LOG.info(ANOTHER_INSTANCE_LOG_INFO);
+    	        if (!MessageDialog.openQuestion(Display.getDefault().getActiveShell(), DIALOG_BOX_TITLE, DIALOG_QUESTION)) {
+    	            shutDown = true;
+    	            PlatformUI.getWorkbench().close();
+    	        }
+    		}
+    		if (!shutDown) {
+    			writeTempFile(new String[] {Long.toString(ProcessHandle.current().pid())}, true);
+    		}
+		} catch (Exception e) {
+			LOG.error(MULTIPLE_INSTANCES_CHECKING_ERROR);
+			e.printStackTrace();
+		}
+    }
+    
+    /**
+     * This function compares contents of temporary file to the list of PIDs obtained from process list
+     * and overwrites the file only with those that match.
+     * @return New contents of temporary file as a string array.
+     * @throws IOException - if any of the operations result in IOException.
+     */
+    String[] clearTempFile() throws IOException {
+    	String[] current = readTempFile();
+    	
+        ArrayList<String> procs = new ArrayList<String>();
+        ProcessHandle.allProcesses()
+        .forEach(process -> procs.add(Long.toString(process.pid())));
+        
+        ArrayList<String> newContent = new ArrayList<String>();
+        
+        for (String l : current) {
+        	for (String s : procs) {
+        		if (l.equals(s)) {
+        			newContent.add(l);
+        			break;
+        		}
+        	}
+        }
+        String[] newContentArr = newContent.toArray(new String[newContent.size()]);
+        writeTempFile(newContentArr, false);
+        
+        return newContentArr;
+    }
+    
+    /**
+     * Locks the temporary file. The lock is only needed in a rare edge case
+     * where someone would fire multiple instances exactly at the same time,
+     * creating write racing conditions.
+     * @return Lock, or empty optional if it fails to acquire lock within reasonable time.
+     * @throws IOException - if any of the operations result in IOException.
+     */
+    @SuppressWarnings("checkstyle:magicnumber")
+    Optional<FileLock> lockTemporaryFile() throws IOException {
```

Comment:
Let's get rid of the locking, the mechanism is brittle and I can't see an easy way around it - unless you can find a less brittle way easily. Trying to anticipate the edge case is great, but I think it's enough of an edge case that we are fine without it